Assets/LeapMotion/Core/Scripts/Attributes/ImplementsInterface.cs

```diff
@@ -55,16 +56,22 @@ public class ImplementsInterfaceAttribute : CombinablePropertyAttribute,
     /// the interface that this attribute constrains objects to, and returns the object
     /// that implements that interface, or null if none was found.
     /// </summary>
-    public UnityObject FindImplementer(UnityObject obj) {
+    public UnityObject FindImplementer(SerializedProperty property, UnityObject obj) {
 
-      if (!fieldInfo.FieldType.IsAssignableFrom(obj.GetType())
-          && !(typeof(Component).IsAssignableFrom(fieldInfo.FieldType)
-               && obj.GetType() == typeof(GameObject))) {
+      // Don't use fieldInfo here because it is very convenient for it to be left null
+      // on the CombinablePropertyAttribute when dealing with generic-type situations
+      // where the ImplementsInterface class has to be constructed manually in a custom
+      // editor. (Specific case: StreamConnectorEditor.cs)
+
+      if (!this.type.IsAssignableFrom(obj.GetType())
```

Comment:
This changes functionality.  Now when I drop a ScriptableObject that does implement the interface, onto a field that requires the interface but is of type Component, the drag-drop reports that the drop should succeed, but it doesn't.  In fact, probing the code shows that the scriptable object is indeed assigned to the serializedProperty, but luckily Unity throws no errors and takes no action.  But that is not something we can guarantee happening for all time (and in fact assigning an object of an invalid type would make sense for an error in the future!)

Reply:
I knew this would break something when I changed it, I just couldn't remember what exactly. :P will fix!

Reply:
and will add a comment because that's what comments are for